config-maven/action.yml

```diff
@@ -0,0 +1,105 @@
+
+outputs:
+  build-number:
+    description: The current build number. Also set as environment variable BUILD_NUMBER
+    value: ${{ steps.get_build_number.outputs.BUILD_NUMBER }}
+  snapshot-version:
+    description: The project version set in the pom.xml (before replacement). Also set as environment variable CURRENT_VERSION
+    value: ${{ steps.config.outputs.snapshot-version }}
+  project-version:
+    description: The project version with build number (after replacement). Also set as environment variable PROJECT_VERSION
+    value: ${{ steps.config.outputs.project-version }}
+
+runs:
+  using: composite
+  steps:
+    - name: Get build number
+      id: get_build_number
+      uses: SonarSource/ci-github-actions/get-build-number@v1
+
+    - name: Set parameter for Vault
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        ARTIFACTORY_READER_ROLE: ${{ inputs.artifactory-reader-role != '' && inputs.artifactory-reader-role ||
+          (github.event.repository.visibility == 'public' && 'public-reader' || 'private-reader') }}
+      run: |
+        echo "ARTIFACTORY_READER_ROLE=${ARTIFACTORY_READER_ROLE}" >> "$GITHUB_ENV"
+
+    - name: Get secrets from Vault
+      id: vault
+      uses: SonarSource/vault-action-wrapper@320bd31b03e5dacaac6be51bbbb15adf7caccc32 # 3.1.0
+      with:
+        secrets: |
+          development/artifactory/token/{REPO_OWNER_NAME_DASH}-${{ env.ARTIFACTORY_READER_ROLE }} username | ARTIFACTORY_USERNAME;
+          development/artifactory/token/{REPO_OWNER_NAME_DASH}-${{ env.ARTIFACTORY_READER_ROLE }} access_token | ARTIFACTORY_ACCESS_TOKEN;
+
+    - name: Set environment variables for Artifactory authentication
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        ARTIFACTORY_USERNAME: ${{ fromJSON(steps.vault.outputs.vault).ARTIFACTORY_USERNAME }}
+        ARTIFACTORY_ACCESS_TOKEN: ${{ fromJSON(steps.vault.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+      run: |
+        echo "ARTIFACTORY_USERNAME=$ARTIFACTORY_USERNAME" >> "$GITHUB_ENV"
+        echo "ARTIFACTORY_ACCESS_USERNAME=$ARTIFACTORY_USERNAME" >> "$GITHUB_ENV"  # deprecated, backward compliance
+        echo "ARTIFACTORY_ACCESS_TOKEN=$ARTIFACTORY_ACCESS_TOKEN" >> "$GITHUB_ENV"
+        echo "ARTIFACTORY_PASSWORD=$ARTIFACTORY_ACCESS_TOKEN" >> "$GITHUB_ENV"  # deprecated, backward compliance
+
+    - name: Set Maven Options
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        MAVEN_OPTS: ${{ inputs.maven-opts }}
+      run: |
+        echo "MAVEN_OPTS=$MAVEN_OPTS" >> "$GITHUB_ENV"
+        echo "maven-opts=$MAVEN_OPTS" >> "$GITHUB_OUTPUT"
+
+    - name: Create Maven command alias with common flags
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        COMMON_MVN_FLAGS: ${{ inputs.common-mvn-flags }}
+      run: |
+        echo "alias mvn='mvn ${COMMON_MVN_FLAGS}'" >> "$HOME/.bash_aliases"
+        echo "BASH_ENV=$HOME/.bash_aliases" >> "$GITHUB_ENV"
```

Comment:
To review live

Reply:
if complex, export COMMON_MVN_FLAGS and developers will have to use it explicitly, or no export